### ISSUE TITLE
path.relative behavior mismatch breaks sveltekit

### ISSUE THREAD

**Issue by ottomated** [oven-sh/bun#4616]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.4.12-arch1-1 x86_64 unknown

### What steps can reproduce the bug?

Run the following code in node and bun:
```js
import { relative } from 'node:path';
console.log(relative('/test', '../b'));
```

### What is the expected behavior?

```
$ pwd
/home/otto/Code/Projects
$ node bun-issue.mjs
../home/otto/Code/b
```

### What do you see instead?

```
$ pwd
/home/otto/Code/Projects
$ bun bun-issue.mjs
../home/otto/Code/Projects/b
```

### Additional information

This breaks Sveltekit inside a monorepo, I believe, because it calculates the wrong relative path for an import.

**Comment by Hanaasagi** [comment]
Related to https://github.com/oven-sh/bun/issues/4789

> This breaks Sveltekit inside a monorepo, I believe, because it calculates the wrong relative path for an import.

Will unblocking sveltekit if the relative path issue is fixed?

**Comment by ottomated** [comment]
Yes, this edge case only happens when a sveltekit is in a subdirectory of a monorepo

**Comment by nektro** [comment]
this looks to be working now as expected. are you able to confirm?

```
❯ pwd
/Users/meghandenny/src/test

❯ bun-debug run ./index.js
../Users/meghandenny/src/b
```

**Comment by ottomated** [comment]
Closed by #4789
